internal/js/modules/k6/browser/browser/frame_mapping.go

```diff
@@ -387,8 +387,15 @@ func mapFrame(vu moduleVU, f *common.Frame) mapping {
 				return nil, fmt.Errorf("parsing frame wait for navigation options: %w", err)
 			}
 
-			return k6ext.Promise(vu.Context(), func() (result any, reason error) {
-				resp, err := f.WaitForNavigation(popts)
+			// Inject JS regex checker for URL pattern matching
+			ctx := vu.Context()
+			jsRegexChecker, err := injectURLMatcherScript(ctx, vu, f.Page().TargetID())
```

Comment:
Out of curiosity, is this code executed every time there's a call to `waitForNavigation`? I guess it's a tradeoff, between this and injecting it only once but always, and I'm not familiar enough with the overhead of injecting scripts, but wondering whether there's another approach that could be slightly better 🤔

Reply:
This is running the JS code into the current Sobek context during runtime. I was wondering if there was a way of doing it once, or if there was an best practices. With out existing APIs (such as `page.on`) we run the JS script every time `page.on` is called.

For the browser module, I think the bottleneck is the JSON marshalling/unmarshalling when it talks to chromium over the websocket, which this isn't using, so maybe we don't need to worry about it today?